vpr/src/place/initial_placement.cpp

```diff
@@ -314,6 +321,69 @@ static t_physical_tile_type_ptr pick_placement_type(t_logical_block_type_ptr log
     return nullptr;
 }
 
+vtr::vector<ClusterBlockId, t_block_score> assign_block_scores() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.placement();
+
+    auto blocks = cluster_ctx.clb_nlist.blocks();
+    auto pl_macros = place_ctx.pl_macros;
+
+    t_block_score score;
+
+    vtr::vector<ClusterBlockId, t_block_score> block_scores;
+
+    for (unsigned int i = 0; i < blocks.size(); i++) {
```

Comment:
just do a resize, easier to read or can say how big you want to be in constructor